src/components/negotiations-panel.tsx

```diff
@@ -0,0 +1,256 @@
+  forfeited: {
+    backgroundColor: theme.palette.warning.main
+  },
+  defunct: {
+    backgroundColor: theme.palette.error.main
+  },
+  unresolved: {
+    backgroundColor: theme.palette.error.main
+  },
+  ongoing: {
+    backgroundColor: "yellow"
+  }
+}));
+
+interface Parameter
+{
+  conflict_version : number;
+  sequence : number[];
+};
+
+export interface NegotiationsPanelProps {
+  conflicts : Readonly<Record<string, NegotiationConflict>>;
+  spotlight?: Readonly<SpotlightValue<string>>;
+  trajManager?: Readonly<RobotTrajectoryManager>;
+  negotiationTrajStore : Record<string, NegotiationTrajectoryResponse>;
+}
+
+export default function NegotiationsPanel(props: NegotiationsPanelProps): JSX.Element {
+  const { conflicts, spotlight, trajManager, negotiationTrajStore } = props;
+
+  React.useEffect(() => {
+    if (!spotlight) {
+      return;
+    }
+    // TODO: spotlight
+  }, [spotlight]);
+  
+  const classes = useStyles();
+
+  /**  Utility conversion functions **/
+  const determineStatusText = (status : NegotiationStatus,
+    parent_resolved : ResolveState): string => {
+    if (status.forfeited)
+      return "  [FORFEITED]";
+    else if (status.rejected)
+      return "  [REJECTED]";
+    else if (status.defunct)
+      return "  [DEFUNCT]";
+    else if (parent_resolved === ResolveState.RESOLVED)
+      return "  [FINISHED]";
+    else
+      return "  [ONGOING]";
+  };
+  const determineStyle = (status : NegotiationStatus, 
+    parent_resolved : ResolveState): string => {
+    if (status.forfeited)
+      return classes.forfeited;
+    else if (status.rejected)
+      return classes.rejected;
+    else if (status.defunct)
+      return classes.defunct;
+    else if (parent_resolved === ResolveState.RESOLVED)
+      return classes.finished;
+    else
+      return classes.ongoing;
+  };
+
+  // keep track of parameters so we can send them as requests
+  let nodeidToParameters = new Map<string, Parameter>();
+
+  /** Render Negotiation panel contents **/
+  const renderNegotiations = (version : string, conflict : NegotiationConflict) => {
+    let conflictLabel = "Conflict #" + version + ", Participants: ";
+    let i = 0;
+    for (const name of Object.values(conflict.participantIdsToNames))
+    {
+      conflictLabel += name;
+
+      if (i !== (Object.keys(conflict.participantIdsToNames).length - 1))
+        conflictLabel += ", ";
+      ++i;
+    }
+    let conflictStyle = classes.ongoing;
+    if (conflict.resolved === ResolveState.RESOLVED)
+      conflictStyle = classes.finished;
+    else if (conflict.resolved === ResolveState.FAILED)
+      conflictStyle = classes.unresolved;
+
+    let tableDom : JSX.Element[] = [];
+    for (const [participantId, statusData] of Object.entries(conflict.participantIdsToStatus))
+    {
+      // status handling and background
+      const participantName = conflict.participantIdsToNames[participantId];
+        
+      //add 1 or 2 rows of data depending on the sequence
+      
```

Comment:
nit: // add
maybe remove extra line?

Reply:
Mmmhh.. I'm seeing this pattern on the comments `//word` (without space) in some places, just out of curiosity do you have a rule to place those?

Reply:
nope. does this repo have a linter i can use?

Reply:
it uses eslint, you need to set up your dev env to use it. However, the linter rules does not check the stylistic things, there is prettier to make sure the styles are right, which should be automatically ran on commit, but it doesn't seem to happen in this case, maybe something is wrong your ide/nodejs?

Reply:
apply prettier 75b99c8